app/controllers/users_controller.rb

```diff
@@ -3,31 +3,14 @@ class UsersController < ApplicationController
 
   def show
     @user = User.find(params[:id])
-  end
+    @dinosaurs = Dinosaur.where(user: @user)
 
-  def edit
-    @user = User.find(params[:id])
     if current_user != @user
       flash[:errors] = "You're not signed in as this user"
       redirect_to dinosaurs_path
     end
   end
 
-  def update
-    @user = User.find(params[:id])
-    if current_user != @user
-      flash[:errors] = "You're not signed in as this user"
-      redirect_to dinosaurs_path
```

Comment:
You could factor this out into a `before_action`, probably. Write a custom method that does this stuff for you to DRY it up? The [example in the docs](http://guides.rubyonrails.org/action_controller_overview.html#filters) might provide some inspiration.